drivers/i2c/Kconfig

```diff
@@ -53,4 +49,8 @@ config I2CMULTIPLEXER_PCA9540BDP
 
 endmenu # I2C Multiplexer Support
 
+config ARCH_HAVE_I2CRESET
```

Comment:
ARCH_HAVE_xxx should exist unconditionally, please see the discussion here:
https://github.com/apache/incubator-nuttx/pull/217
https://github.com/apache/incubator-nuttx/pull/228
https://github.com/apache/incubator-nuttx/pull/232

Reply:
Hi Xiang,
I've pulled ARCH_HAVE_I2CRESET out of the if I2C conditional. The conditional must be first in this file as menuconfigs will only nest configs if the conditional is the next block after declaring the menuconfig.